src/fairchem/core/components/calculate/solvation_runner.py

```diff
@@ -0,0 +1,146 @@
+"""
+Copyright (c) Meta Platforms, Inc. and affiliates.
+
+This source code is licensed under the MIT license found in the
+LICENSE file in the root directory of this source tree.
+"""
+
+from __future__ import annotations
+
+import os
+import pickle
+from typing import TYPE_CHECKING, Any, ClassVar
+
+import numpy as np
+import pandas as pd
+from tqdm import tqdm
+
+from fairchem.core.components.calculate import CalculateRunner
+
+if TYPE_CHECKING:
+    from ase.calculators.calculator import Calculator
+
+
+class SolvationRunner(CalculateRunner):
+    """
+    Singlepoint evaluator for OC25 solvation energy. #to-do: write an explanation
+    """
+
+    result_glob_pattern: ClassVar[str] = "solvation_*-*.json.gz"
+
+    def __init__(self, calculator: Calculator, input_data_path: str):
+        """
+        Initialize the SolvationRunner
+
+        Args:
+            calculator: ASE calculator to use for energy calculations
+            input_data_path (dict): Path to dataset containing atoms object for solvation inputs
+        """
+        with open(input_data_path, "rb") as f:
+            input_data = pickle.load(f)
+        super().__init__(calculator=calculator, input_data=input_data)
+
+    def calculate(self, job_num: int = 0, num_jobs: int = 1) -> list[dict[str, Any]]:
+        """
+        Perform solvation energy calculations on the input data.
+
+        Args:
+            job_num: Job number for parallel processing
+            num_jobs: Total number of jobs for parallel processing
+
+        Returns:
+            list[dict[str, Any]] containing results for each key in the input data
+        """
+        all_results = []
+        chunk_indices = np.array_split(range(len(self.input_data)), num_jobs)[job_num]
+        for i in tqdm(
+            chunk_indices, desc="Calculating ground truth and running single points"
+        ):
+            key = list(self.input_data.keys())[i]
+            # define all the atoms for calculator set up
+            randsh0 = self.input_data[key]["randsh0"]
+            randsh0_surface = self.input_data[key]["randsh0_surface"]
```

Comment:
i know the `randsh0` naming is rather ugly, can we at least change the variable names to be straightforward. i.e. `solvated_interface`, `solvated_surface`, `vacuum_interface`, `vacuum_surface`